packages/arb-token-bridge-ui/src/util/TokenDepositUtils.ts

```diff
@@ -0,0 +1,53 @@
+import { getL2Network } from '@arbitrum/sdk'
+import { Inbox__factory } from '@arbitrum/sdk/dist/lib/abi/factories/Inbox__factory'
+import { Provider } from '@ethersproject/providers'
+import { BigNumber } from 'ethers'
+import { DepositGasEstimates } from '../hooks/arbTokenBridge.types'
+
+export async function depositTokenEstimateGas({
+  l1Provider,
+  l2Provider
+}: {
+  l1Provider: Provider
+  l2Provider: Provider
+}): Promise<DepositGasEstimates> {
+  const l1BaseFee = await l1Provider.getGasPrice()
+  const l2Network = await getL2Network(l2Provider)
```

Comment:
NIT: `l2Network` doesn't rely on `l1BaseFee`, we can use `Promise.all` here to be slightly faster

```
  const [l1BaseFee, l2Network] = await Promise.all([
    l1Provider.getGasPrice(),
    getL2Network(l2Provider)
  ])
```